qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -262,22 +287,28 @@ def compute_eigenvalues(
             start_time = time()
 
             # TODO: add gradient support after FidelityGradients are implemented
-            if callable(self.optimizer):
-                opt_result = self.optimizer(  # pylint: disable=not-callable
+            if isinstance(self.optimizer, list):
+                optmzer = self.optimizer[step - 1]
```

Comment:
You don't have to change this but would it be better legible to use `optimizer` over `optmzr`? 😬